File: backend/src/main/scala/domains/problem/objects/request/DeleteProblemDataPathRequest.scala
package domains.problem.objects.request

import domains.problem.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

/** 删除题目数据路径的请求体；path 可以指向单文件或目录前缀。 */
final case class DeleteProblemDataPathRequest(path: ProblemDataPath)

/** DeleteProblemDataPathRequest 的 JSON 编解码器。 */
object DeleteProblemDataPathRequest:
  given Encoder[DeleteProblemDataPathRequest] = deriveEncoder[DeleteProblemDataPathRequest]
  given Decoder[DeleteProblemDataPathRequest] = deriveDecoder[DeleteProblemDataPathRequest]

File: frontend/src/objects/problem/request/DeleteProblemDataPathRequest.ts
import type { ProblemDataPath } from '@/objects/problem/ProblemDataPath'

/** 删除题目数据路径请求体；path 必须是已校验的相对数据路径。 */
export type DeleteProblemDataPathRequest = {
  path: ProblemDataPath
}
